### PR TITLE
Add partial arg to parse_GPL and get_GEO. #53

### PR DESCRIPTION
**Pull request by ttyskg**
Issue #53 
Add a 'partial' argument to parse_GPL() and get_GEO() to parse GSMs
partially from large GPL file. This 'partial' argument is an iterable
object containing the GSM accession IDs that you want to parse
partially from the GPL.

### CODE REVIEW

File: GEOparse/GEOparse.py [hunk 1]
```diff
@@ -421,6 +425,8 @@ def parse_GPL(filepath, entry_name=None):
                         gses[entry_name] = GSE(name=entry_name,
                                                metadata=gse_metadata)
                     elif entry_type == "SAMPLE":
+                        if (partial) and (entry_name not in partial):
```
Comment (guma44) [inline GEOparse/GEOparse.py]: I would drop not necessary parenthesis.

File: GEOparse/GEOparse.py [hunk 2]
```diff
@@ -388,14 +390,16 @@ def parse_GSM(filepath, entry_name=None):
...
-def parse_GPL(filepath, entry_name=None):
+def parse_GPL(filepath, entry_name=None, partial=None):
     """Parse GPL entry from SOFT file.
 
     Args:
         filepath (:obj:`str` or :obj:`Iterable`): Path to file with 1 GPL entry
             or list of lines representing GPL from GSE file.
         entry_name (:obj:`str`, optional): Name of the entry. By default it is
             inferred from the data.
+        partial (:obj:'iterable', optional): Accession ID of GSM that you want
```
Comment (guma44) [inline GEOparse/GEOparse.py]: For the help I would go for something like that: A list of accession IDs of GSMs to be partially extracted from GPL, works only if a file/accession is a GPL?

**Comment by guma44** [comment]
There is something wrong with checks - especially with Travis CI. It has some problems with downloads.

File: GEOparse/GEOparse.py (hunk 2 above)
Comment (ttyskg) [reply GEOparse/GEOparse.py]: Yours explain of 'partial' argument is more evident than mine. I will update the explains.

**Comment by ttyskg** [comment]
I modified the part you commented. Although Travis CI failed, my local tests all succeeded.

I'm sorry, I can not determine the cause of the Travis CI failure.

**Comment by guma44** [comment]
No problem. This more of the issue with the Travis. I will look into it when I have time but rather later than sooner as the local tests all go well as you noticed.

**Pull request merged**